src/ext/language_tools.js

```diff
@@ -182,11 +198,33 @@ require("../config").defineOptions(Editor.prototype, "editor", {
                 // On each change automatically trigger the autocomplete
                 this.commands.on('afterExec', doLiveAutocomplete);
             } else {
-                this.commands.removeListener('afterExec', doLiveAutocomplete);
+                this.commands.off('afterExec', doLiveAutocomplete);
             }
         },
         value: false
     },
+    liveAutocompletionDelay: {
+        set: function (val) {
+            if (val) {
```

Comment:
Why do we need this condition? Does it mean that if we set the delay to X initially, we then won't be able to set it to 0 at runtime? Same question for the threshold

Reply:
We did not need the set/get methods at all, removed them now.